policyengine_api/country.py

```diff
@@ -483,3 +488,40 @@ def validate_country(country_id: str) -> Union[None, Response]:
         )
         return Response(json.dumps(body), status=404)
     return None
+
+# Write a utility function to take a given variable name and recurse through its adds and subtracts values and append each given variable to an array; 
+# this function will also need to handle list parameters, which can be accessed through the system’s get_parameter function, I believe, 
+# and can be tested for by checking if the adds is of type str
+
+def get_all_variables(variable_name: str, parameter_name: str, system: TaxBenefitSystem) -> list:
+    """
+    Get all variables that are added or subtracted to the given variable name.
+
+    Args:
+        variable_name (str): The variable name.
+        parameter_name (str): The parameter name.
+        system (TaxBenefitSystem): The tax benefit system.
+
+    Returns:
+        list: The list of variable names.
+    """
+    variable = system.get_variable(variable_name)
+    adds = variable.adds
+    subtracts = variable.subtracts
+    variables = []
+
+    for add in adds:
+        if isinstance(add, str):
+            variables.append(add)
+        elif isinstance(add, list):
+            for item in add:
+                variables.append(item)
```

Comment:
I believe your implementation here will append the entire array, but won't recurse further into `add` items further down the tree. I think you'd want to write a recursive function here that, when there is an adds and/or a subtracts, calls `get_all_variables` on that next tier downward, until eventually you hit some marker of there being no more levels.